File: 145.c
#include <stdio.h>

// reverse number
int 
rev(int n)
{
	int v = 0;
	while( n > 0 ){
		v = 10 * v + n % 10;
		n /= 10;
	}
	return v;
}

// check every digit is Odd
int 
isOddDigit(int n)
{
	int r;
	while (n != 0) {
		r = n % 10;
		if (!( r & 0x1))// is this digit is even, directly return
			return 0;
		n /= 10;
	}
	return 1;
}

int
main()
{
	int i;
	int cnt = 0;
	int thresh = 1000 * 1000 * 1000;
	for (i=0; i< thresh; i++) {
		if (i % 10 == 0)// skip leading zero in reverse number
			continue;
		cnt += isOddDigit(i + rev(i));
	}
	printf("%d\n",cnt);
	return 0;
}

File: 145.py
def rev(n):
    v = 0
    while n > 0:
        v = v * 10 + n % 10
        n = n / 10
    return v


oddset = frozenset([1,3,5,7,9])
def isOddDigit(n):
    while n:
        r = n % 10
        if r not in oddset:
            return False
        n = n / 10
    return True


N = 10 ** 9
cnt = 0
for n in xrange(1, N):
    if n % 10 == 0:
        continue
    if isOddDigit(n + rev(n)):
        cnt += 1
print cnt
